chainservice/chainservice.go

```diff
@@ -378,19 +378,31 @@ func New(
 
 			return nil
 		},
-		func(ctx context.Context, start uint64, end uint64) error {
+		func(ctx context.Context, start uint64, end uint64, repeat int) {
 			peers, err := p2pAgent.Neighbors(ctx)
 			if err != nil {
-				return err
+				log.L().Error("failed to get neighbours", zap.Error(err))
+				return
 			}
 			if len(peers) == 0 {
-				return errors.New("no peers")
+				log.L().Error("no peers")
+			}
+			if repeat < 2 {
+				repeat = 2
+			}
+			if repeat > len(peers) {
+				repeat = len(peers)
+			}
+			for i := 0; i < repeat; i++ {
```

Comment:
why we want to send same request to >= 2 peers?
do we have test/data to indicate there's a high probability of sync request not received/served by peers? my concern is that this will increase (unnecessary) data traffic on the p2p network

Reply:
The ideal case would be `adaptive`, e.g., request from 1, if doesn't work, request from 2; then 4; 8, 16...

2 seems a good start. we can get this one onto testnet and see how it works.